blockchain/chain.go

```diff
@@ -511,6 +511,52 @@ func (b *BlockChain) getPrevNodeFromNode(node *blockNode) (*blockNode, error) {
 	return prevBlockNode, err
 }
 
+// ancestorNode returns the ancestor block 'distance' blocks before the passed
+// anchor block. While iterating backwards through the chain, any block nodes
+// which aren't in the memory chain are loaded in dynamically.
+//
+// This function MUST be called with the chain state lock held (for writes).
+func (b *BlockChain) ancestorNode(anchor *blockNode, distance uint32) (*blockNode, error) {
```

Comment:
I've renamed the function to `relativeNode`. Funnily, there's a bit of a double-entendre w.r.t the function name as it retrieves a node which is a relative of the ancestor node by walking back a relative distance into the chain 😎.